### PR TITLE
MGDOBR-950: Create dns entry during the provisioning of the dependencies for a bridge

### PR DESCRIPTION
**Pull request by r00ta**
[MGDOBR-950](https://issues.redhat.com/browse/MGDOBR-950)

The feature is not exposed yet to the user. We only create and delete the dns entry but it's not used yet. 

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] The layers in the `kustomize` folder have been updated accordingly.
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

**Review by jiripetrlik — APPROVED** [review]

File: manager/src/main/java/com/redhat/service/smartevents/manager/workers/resources/BridgeWorker.java [hunk 1]
```diff
@@ -75,6 +79,9 @@ public Bridge createDependencies(Work work, Bridge bridge) {
         // We don't need to wait for the Bridge to be READY to create the Error Handler.
         createErrorHandlerProcessor(bridge);
 
+        // Create DNS record
+        dnsService.createDnsRecord(bridge.getId());
```
Comment (danielezonca) [inline manager/src/main/java/com/redhat/service/smartevents/manager/workers/resources/BridgeWorker.java:83]: This method adds up to 5 seconds to the execution time (in the worst case scenario).
As far as I remember this method is now invoked by Quartz and I don't know if this additional time can reach some external/end-to-end transaction timeout 🤔

**Review by danielezonca — APPROVED** [review]
LGTM, just a comment about making sure it has no impact on the Quartz scheduling side

File: manager/src/main/java/com/redhat/service/smartevents/manager/workers/resources/BridgeWorker.java (hunk 1 above)
Comment (r00ta) [reply manager/src/main/java/com/redhat/service/smartevents/manager/workers/resources/BridgeWorker.java:83]: afaik quartz has no timeout on the jobs (it's not even possible to configure it)

File: manager/src/main/java/com/redhat/service/smartevents/manager/workers/resources/BridgeWorker.java [hunk 2]
```diff
@@ -75,6 +80,9 @@ public Bridge createDependencies(Work work, Bridge bridge) {
         // We don't need to wait for the Bridge to be READY to create the Error Handler.
         createErrorHandlerProcessor(bridge);
 
+        // Create DNS record
+        dnsService.get().createDnsRecord(bridge.getId());
```
Comment (manstis) [inline manager/src/main/java/com/redhat/service/smartevents/manager/workers/resources/BridgeWorker.java:84]: Did you want to make your `DnsService` implementations `@ApplicationScoped`?

I think the use of `get()` will return a new instance on each invocation as they have the default scope.

Comment (r00ta) [reply manager/src/main/java/com/redhat/service/smartevents/manager/workers/resources/BridgeWorker.java:84]: Wops, that's a leftover

**Review by manstis — APPROVED** [review]

File: app-interface/manifests/rhose-fleet-manager.yml [hunk 3]
```diff
@@ -273,6 +273,8 @@ objects:
                       key: EVENT_BRIDGE_DNS_SUBDOMAIN_TLS_KEY
                 - name: EVENT_BRIDGE_DNS_SUBDOMAIN
                   value: .bf2.smartevents-stage.rhcloud.redhat.com
+                - name: EVENT_BRIDGE_K8S_ORCHESTRATOR
```
Comment (r00ta) [inline app-interface/manifests/rhose-fleet-manager.yml:276]: @msmagnanijr I've updated this template fyi

**Review by jiripetrlik — APPROVED** [review]

**Review by danielezonca — APPROVED** [review]
LGTM :)

**Pull request merged**